muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/organizationmanagement/rest/OrganizationUserManagementRESTService.java

```diff
@@ -171,6 +172,69 @@ public Response searchStaffMembers(
     return Response.ok(responseStaffMembers).build();
   }
   
+  @GET
+  @Path("/studentsSummary")
+  @RESTPermit(OrganizationManagementPermissions.ORGANIZATION_VIEW)
+  public Response studentsSummary(
+      @QueryParam("firstResult") @DefaultValue("0") Integer firstResult,
+      @QueryParam("maxResults") @DefaultValue("10") Integer maxResults) {
+    
+    if (!sessionController.isLoggedIn()) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+
+    SearchProvider searchProvider = searchProviderInstance.get();
+    if (searchProvider == null) {
+      return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+    }
+
+    UserSchoolDataIdentifier userSchoolDataIdentifier = userSchoolDataIdentifierController
+        .findUserSchoolDataIdentifierBySchoolDataIdentifier(sessionController.getLoggedUser());
+    OrganizationEntity organization = userSchoolDataIdentifier.getOrganization();
+
+    SearchResult result = searchProvider.searchUsers(
+        Arrays.asList(organization),
+        "",
+        null,                                                 // fields
+        Arrays.asList(EnvironmentRoleArchetype.STUDENT),
+        null,                                                 // userGroupFilters
+        null,                                                 // workspaceFilters
+        null,                                                 // userIdentifiers
+        true,                                                 // includeInactiveStudents
+        true,                                                 // includeHidden
+        true,                                                 // onlyDefaultUsers
+        firstResult,
+        maxResults);
+
+    List<Map<String, Object>> results = result.getResults();
+    OrganizationStudentsActivityRESTModel studentActivityRESTModel = null;
+    if (results != null && !results.isEmpty()) {
+      List<UserEntity> activeStudents = new ArrayList<>();
+      List<UserEntity> inActiveStudents = new ArrayList<>();
```

Comment:
UserEntities themselves aren't needed at all, so no need to waste time or resources fetching and storing them. These could be replaced with just...

```
int activeStudentCount = 0;
int inactiveStudentCount = 0;
```